File: android-utils/src/main/java/com/example/android/utils/widgets/TransparentProgressDialog.java
package com.example.android.utils.widgets;

import android.app.Dialog;
import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.annotation.ColorRes;
import android.support.annotation.NonNull;
import android.support.annotation.StyleRes;
import android.support.v4.content.ContextCompat;
import android.view.Gravity;
import android.view.WindowManager;
import android.widget.RelativeLayout;

import com.example.android.R;
import com.example.android.databinding.ComponentProgressDialogBinding;

public class TransparentProgressDialog extends Dialog {
	public TransparentProgressDialog(@NonNull Context context, @StyleRes int style, @ColorRes int color) {
		super(context, style);

		if (getWindow() == null) {
			return;
		}

		WindowManager.LayoutParams layoutParams = getWindow().getAttributes();
		layoutParams.gravity = Gravity.CENTER_HORIZONTAL;

		getWindow().setAttributes(layoutParams);

		setTitle(null);
		setCancelable(false);
		setOnCancelListener(null);

		ComponentProgressDialogBinding binding =
			DataBindingUtil.inflate(getLayoutInflater(), R.layout.component_progress_dialog, null, false);

		binding.loading.getIndeterminateDrawable().setColorFilter(ContextCompat.getColor(getContext(), color), android.graphics.PorterDuff.Mode.MULTIPLY);
		addContentView(binding.getRoot(), new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT));
	}
}

File: libs/toolkit-android/src/main/kotlin/com/example/utils/widgets/TransparentProgressDialog.kt
package com.example.utils.widgets

import android.app.Dialog
import android.content.Context
import android.support.annotation.ColorRes
import android.support.annotation.StyleRes
import android.support.v4.content.ContextCompat
import android.view.Gravity
import android.widget.RelativeLayout
import com.example.android.R
import kotlinx.android.synthetic.main.component_progress_dialog.*

class TransparentProgressDialog(context: Context, @StyleRes style: Int, @ColorRes color: Int) : Dialog(context, style) {
  init {
    val layoutParams = window!!.attributes
    layoutParams.gravity = Gravity.CENTER_HORIZONTAL

    window!!.attributes = layoutParams

    setTitle(null)
    setCancelable(false)
    setOnCancelListener(null)

    setContentView(R.layout.component_progress_dialog)

    loading.indeterminateDrawable.setColorFilter(ContextCompat.getColor(getContext(), color), android.graphics.PorterDuff.Mode.MULTIPLY)
    addContentView(root, RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT))
  }
}
